presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -383,7 +395,19 @@ public ActualProperties visitJoin(JoinNode node, List<ActualProperties> inputPro
                     constants.putAll(probeProperties.getConstants());
                     constants.putAll(buildProperties.getConstants());
 
+                    // Cross join only preserves constant(probe_column) local properties.
+                    // Cross join doesn't preserve sorting(probe_column) or grouping(probe_column) local properties.
+                    List<LocalProperty<Symbol>> newLocalProbeProperties;
+                    if (node.isCrossJoin()) {
+                        Set<Symbol> probeConstants = LocalProperties.extractLeadingConstants(probeProperties.getLocalProperties());
```

Comment:
@sopel39 I've been thinking that a given node should keep only properties that it is guaranteed to preserve. Currently, there are 3 types of local properties: constant, sorting and grouping. Cross join preserves only constant properties on probe and build sides. It doesn't preserve sorting or grouping on either side. Hence, this code explicitly keeps only leading constants on the probe side. (I'm seeing now that it should preserve leading constants on the build side as well. However, as you pointed out this happens automatically in the constructor. Therefore, I'll update the code to strip *all* local properties.) If we add a new type of local property in the future, this code will continue to be *correct*, but will present an opportunity for optimization if the join happens to preserve the new type of local property. 

I think stripping just the grouping property is not sufficient, because sorting properties are not preserved as well.

Hash inner join preserves all local properties of the probe side.

Reply:
> Therefore, I'll update the code to strip all local properties.) If we add a new type of local property in the future, this code will continue to be correct, but will present an opportunity for optimization if the join happens to preserve the new type of local property.

+1